### PR TITLE
Split RX from RXY

### PR DESCRIPTION
**Pull request by alecandido**
Unfortunately, the `RXY` name is non-standard, but I was lacking a better one.
Any suggestion? @stavros11 @andrea-pasquale 

- ~~restore some validation?~~

### CODE REVIEW

**Comment by andrea-pasquale** [comment]
> Unfortunately, the `RXY` name is non-standard, but I was lacking a better one. Any suggestion? @stavros11 @andrea-pasquale

I was discussing with @stavros11 about this yesterday. What about just `R` or `Rn`?

**Comment by alecandido** [comment]
> I was discussing with @stavros11 about this yesterday. What about just `R` or `Rn`?

To be fair, I'm not even completely sure that it should be a native. I see it better as a compilation rule, since the native should just store the calibrated pulse.

However, by now I'm entering more and more in feature freeze mode. If there is a sufficient agreement about the above statement, it will become an issue for 0.3, and that's it (this "parametrized pi-pulse" is there since months, and I won't change the last day before release).

For me, all names are good enough, just tell me your favorite (one for which there is some consensus), and I can change it.

**Comment by stavros11** [comment]
> I was discussing with @stavros11 about this yesterday. What about just `R` or `Rn`?

I would say `R` is fine. It is simple and doesn't really introduce new physics.

> To be fair, I'm not even completely sure that it should be a native. I see it better as a compilation rule, since the native should just store the calibrated pulse.

I actually think the proper solution is this, have only the `RX` (with fixed amplitude and zero phase) as native. However, the side-effect is that then the user (mainly qibocal) would have to do the pydantic gymnastics (tune the amplitude, change the phase) to get the other rotations. Which is not necessarily bad, we just need to get used to it.

Generally, I would only call natives what we are actually calibrating. If we are calibrating the RX90 directly I would then call it native, but unfortunately I don't think we have support for that in 0.2, native kinds are fixed. If we need to further manipulations to get other gates, I would do them outside the `Platform`.

**Comment by alecandido** [comment]
> I would say `R` is fine. It is simple and doesn't really introduce new physics.

Ah, wait, maybe that name is the one already used:
https://docs.quantum.ibm.com/api/qiskit/qiskit.circuit.library.RGate

**Comment by alecandido** [comment]
> I actually think the proper solution is this, have only the `RX` (with fixed amplitude and zero phase) as native. However, the side-effect is that then the user (mainly qibocal) would have to do the pydantic gymnastics (tune the amplitude, change the phase) to get the other rotations. Which is not necessarily bad, we just need to get used to it.
> 
> Generally, I would only call natives what we are actually calibrating. If we are calibrating the RX90 directly I would then call it native, but unfortunately I don't think we have support for that in 0.2, native kinds are fixed. If we need to further manipulations to get other gates, I would do them outside the `Platform`.

I perfectly agree with you, but I'd also like to save whichever user the gymnastics you mentioned.
Since we already have a proposal to expose the compiler in a smoother way #1042, let's first do that, and then we will reconsider also this method.

**Comment by alecandido** [comment]
> - ~restore some validation?~

There is no strong reason why single qubit gates should be single pulses or applied on individual channels.
True, `RX` is usually the pi-pulse. But if any user wanted to add something else, it's not necessary to deny it.

Let's just leave it to users' responsibility.

File: src/qibolab/_core/native.py [hunk 1]
```diff
@@ -1,92 +1,46 @@
...
 
 
-class FixedSequenceFactory(Native):
-    """Simple factory for a fixed arbitrary sequence."""
+def rxy(seq: PulseSequence, theta: float = np.pi, phi: float = 0.0) -> PulseSequence:
+    """Create a sequence for single-qubit rotation.
 
-    def create_sequence(self) -> PulseSequence:
-        return deepcopy(self)
+    ``theta`` will be the angle of the rotation, while ``phi`` the angle that the rotation axis forms with x axis.
```
Comment (stavros11) [inline src/qibolab/_core/native.py:35]: In principle, we could leave the `Args:` structure in the docstring here, as it was before.

**Review by stavros11 — APPROVED** [review]
Thanks @alecandido, looks good to me.

File: src/qibolab/_core/native.py (hunk 1 above)
Comment (alecandido) [reply src/qibolab/_core/native.py:35]: Yes, we could.

This is something I'm doing on my own: frequently the `Args:` (and equivalents) contain redundant information (essentially repeating the variable names), and redundant type hints.

It was not the case of this docstring in particular, as it is essentially unchanged. But I'm trying to keep the style more discursive.
However, I'm not pushing to enforce it. Not now, at least.

**Comment by alecandido** [comment]
@andrea-pasquale is it fine for you like this?

File: src/qibolab/_core/native.py [hunk 2]
```diff
@@ -1,92 +1,46 @@
...
+    """Normalize theta to (-pi, pi], and phi to [0, 2*pi)."""
+    theta = theta % (2 * np.pi)
+    theta = theta - 2 * np.pi * (theta > np.pi)
+    phi = phi % (2 * np.pi)
+    return theta, phi
 
 
-class FixedSequenceFactory(Native):
-    """Simple factory for a fixed arbitrary sequence."""
+def rxy(seq: PulseSequence, theta: float = np.pi, phi: float = 0.0) -> PulseSequence:
```
Comment (andrea-pasquale) [inline src/qibolab/_core/native.py:32]: At this point we could also call this `r` no?

**Review by andrea-pasquale — APPROVED** [review]
Yes, fine by me. Just a tiny suggestions.

File: src/qibolab/_core/native.py (hunk 2 above)
Comment (alecandido) [reply src/qibolab/_core/native.py:32]: Ah, yes, you're right.

But this is private, so we can even change over and over :P

Comment (alecandido) [reply src/qibolab/_core/native.py]: Done in 195f9ce

**Comment by alecandido** [comment]
I'm merging as soon as the CI will pass

**Pull request merged**